profile/serializers.py

```diff
@@ -0,0 +1,144 @@
+from django.contrib.auth import authenticate
+from django.contrib.auth import get_user_model
+from django.utils.translation import ugettext_lazy as _
+from django.conf import settings
+from rest_framework import exceptions, serializers
+from allauth.utils import get_username_max_length, email_address_exists
+from allauth.account.adapter import get_adapter
+from allauth.account.utils import setup_user_email
+from allauth.account import app_settings as allauth_settings
+from .models import User
+
+UserModel = get_user_model()
+
+
+class UserSerializer(serializers.HyperlinkedModelSerializer):
```

Comment:
Why are you using HyperlinkedModelSerializer?

This generates a url field for the related models instead of a Foreign Key, but you do not seem to have any related model.

Reply:
I totally agree. At the beginning, I was using a one to one relation between Profile model and Django User model. After changing this for a custom User model instead of using the Django User I forgot to update the serializer.